### PR TITLE
[DOC] create a howto for testing pull requests

### PR DESCRIPTION
**Pull request by efx**
Briefly document how to link ember data in your application. Came up as a result of someone wanting to helpfully test an outstanding PR.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -12,6 +12,29 @@ We have provided an [issue template](.github/bug.md) what will help guide you th
 If you are unsure if something is a bug, the `#ember-data` channel on [Discord](https://discord.gg/zT3asNS) is
 a great place to ask for help!
 
+#### Testing ember data source directly
```
Comment (efx) [inline CONTRIBUTING.md]: this is for depending on monolithic ember-data but not individual packages. It does solve the transitive dependency problem.

Comment (efx) [reply CONTRIBUTING.md]: updates before merging
- generalize the language for individual packages
- yarn + npm do not mix so we need to caveat and describe this better (see ember.js discussion)

**Comment by MelSumner** [comment]
@efx it looks like there are a couple of failing tests. Can you look to figure out why?

**Comment by efx** [comment]
@MelSumner certainly! I'm swamped with an internal release so will need to take a look at this after 2/3. I plan to revise the verbiage per Rob's suggestions and will dig into whether the CI failures are legitimate.

**Pull request merged**